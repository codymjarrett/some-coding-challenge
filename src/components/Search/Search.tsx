import React, { useContext, useState, ChangeEvent, MouseEvent } from 'react'
import { PARTIAL_MATCH, RESET } from '../../constants'
import { MovieContext } from '../../context/context'
import movieDataService from '../../services/movieDataService'

const Search: React.FC = () => {
	const { dispatch } = useContext(MovieContext)
	const [input, setInput] = useState('')

	const handleOnChange = (ev: ChangeEvent<HTMLInputElement>) => {
		let substring = ev.currentTarget.value.toLowerCase()
		setInput(substring)
	}

	const handleSearch = (ev: MouseEvent) => {
		dispatch({
			type: PARTIAL_MATCH,
			substring: input,
		})
		setInput('')
	}
	const handleReset = (ev: MouseEvent) => {
		resetState()
	}

	const fetchMovieService = () => {
		return movieDataService.get()
	}

	const resetState = () => {
		const initialState = fetchMovieService()
		initialState.then(data => {
			setInput('')
			if (Array.isArray(data)) {
				dispatch({
					type: RESET,
					payload: data,
				})
			}
		})
	}

	return (
		<div className="search">
			<div className="search__search-bar">
				<input
					placeholder="search for movie..."
					type="text"
					value={input}
					onChange={handleOnChange}
				/>
			</div>
			<div className="search__buttons">
				<button
					className="search__button button"
					type="button"
					onClick={handleSearch}
				>
					Search
				</button>
				<button
					className="search__button button"
					type="button"
					onClick={handleReset}
				>
					Reset
				</button>
			</div>
		</div>
	)
}

export default Search
