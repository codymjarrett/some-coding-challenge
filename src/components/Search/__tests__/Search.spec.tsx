import React from 'react'
import { render, cleanup } from '@testing-library/react';

import Search from '../Search'

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Search/>);
  expect(asFragment()).toMatchSnapshot();
});

