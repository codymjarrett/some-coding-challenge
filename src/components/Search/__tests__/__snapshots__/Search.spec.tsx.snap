// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <div
      class="search__search-bar"
    >
      <input
        placeholder="search for movie..."
        type="text"
        value=""
      />
    </div>
    <div
      class="search__buttons"
    >
      <button
        class="search__button button"
        type="button"
      >
        Search
      </button>
      <button
        class="search__button button"
        type="button"
      >
        Reset
      </button>
    </div>
  </div>
</DocumentFragment>
`;
